src/charm.py

```diff
@@ -1625,6 +1651,12 @@ def _was_restore_successful(self, container: Container, service: ServiceInfo) ->
             logger.debug("Restore check early exit: Patroni has not started yet")
             return False
 
+        try:
+            self._setup_users()
+        except Exception as e:
+            logger.exception(e)
```

Comment:
`logger.exception()` will automatically log thee traceback from the current context. We should provide a log message to give context instead.

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/1049/commits/511fdcea761bc6a748ab580dd75d9c0bc2145c9c.